server/core/runtime/apply_step_runner.go

```diff
@@ -177,6 +178,8 @@ func (a *ApplyStepRunner) runRemoteApply(
 		}
 	}
 
+	close(inCh)
```

Comment:
Is there a reason to close manually vs deferring when it's created?

Reply:
just being explicit as to when its used in that giant function.

Reply:
the only concern here would be if process panics before line, the channel will never closes. But at that point it will probably be garbage collected?

Reply:
right, I'll just change it to defer